BCStudentSoftwareDevTeam/celts
# Add ability to limit event RSVP and add a RSVP waitlist

## This fixes issue #748 : Combined with another issue, now in PR #879
CELTS would like to be able to limit the amount of people that can RSVP to an event. 

## Did: 
Added a new column to the Event table `rsvpLimit` and created a function `getCurrentRsvpAmmount()` in events.py to get the amount of people that have RSVPd to an event. The RSVP button is enabled/disabled based off the current RSVP amount. The wording on the Event View page now changes based off of whether or not RSVP is required, if there is a limit or not, and how many spots are left if there is a limit. 

## To Test:
- Checkout the branch `limitRsvp`
- Reset your database with `database/reset_database.sh test`
- Run the test suite with `tests/run_tests.sh ` and verify that all tests pass
- Run the application with `flask run` 
- Create multiple events, some that require RSVP but have no limit and some that require RSVP with a limit
- Verify that all the behavior that you encounter is as you expect
